### PR TITLE
Fix response when trying to cancel a completed task

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #6465
https://pulp.plan.io/issues/6465

### CODE REVIEW

File: pulpcore/app/viewsets/task.py [hunk 1]
```diff
@@ -79,8 +79,10 @@ def partial_update(self, request, pk=None, partial=True):
         if request.data["state"] != "canceled":
             raise ValidationError(_("The only acceptable value for 'state' is 'canceled'."))
         task = cancel_task(task.pk)
+        # Check whether task is actually canceled
+        http_status = None if task.state == TASK_STATES.CANCELED else status.HTTP_409_CONFLICT
         serializer = self.serializer_class(task, context={"request": request})
-        return Response(serializer.data)
+        return Response(serializer.data, status=http_status)
```
Comment (dralley) [inline pulpcore/app/viewsets/task.py:85]: So if ```http_status=None```, it defaults to 200?

Comment (dralley) [reply pulpcore/app/viewsets/task.py:85]: @bmbouter What should we do if the user tries to cancel a task that is already in canceled state?

Comment (bmbouter) [reply pulpcore/app/viewsets/task.py:85]: I expect pulp not to error. The user can specify cancel to the same task over and over without error.

Comment (dralley) [reply pulpcore/app/viewsets/task.py:85]: Cool, then this is the correct behavior.

**Pull request merged**